app/bingey-ui/src/utils/filter.js

```diff
@@ -0,0 +1,23 @@
+export const filter = (values, filters) => {
+  return values?.filter((value) => isValid(value, filters));
+};
+
+const isValid = (value, filters) => {
+  let filter;
+
+  for (let i = 0; i < filters.length; i++) {
+    filter = filters[i];
+
+    if (Array.isArray(value[filter.field])) {
+      if (!value[filter.field].includes(filter.value)) {
```

Comment:
I imagine a filter would look like this `{field: 'genre', values: ['action', 'horror']}` this seems like a filter can only have one value

Reply:
Yes, I designed it in such a way, that each filter value needs it's own _filter object_ in the `filters` array. Otherwise comparing filter value to an array type field makes this more complicated. With your example, the `filters` array should be 
```
[
  {
    'field': 'genre',
    'value': 'action'
  },
  {
    'field': 'genre',
    'value': 'horror'    
  }
]
```